tests/strategies/test_generic_tests.py

```diff
@@ -0,0 +1,298 @@
+"""Generic tests for all strategies."""
```

Comment:
The name seems a bit weird, maybe should be test_generic_functions?

Reply:
I suppose. The intended name is "Generic tests", but pytest requires all test files to start with `test_`.
I guess it could be `test_all_strategies` instead?

Reply:
yes that would be much better!